### PR TITLE
UI changes to the recommended payment options

### LINKED ISSUE #32130: Payment settings - UI changes to the recommended payment options section
**Description:** we want to introduce some tweaks to this section of the payment settings screen, in order to reduce cognitive load and increase the visibility of WCPay.

**Acceptance criteria:**
- [ ] If the store is located in a country where WCPayments is eligible, hide the **Recommended Ways to Get Paid** panel and only show the "other payment methods" CTA at the bottom of the table.

### Recommended Ways to Get Paid Panel

image

☝️ Hide this when the store is located in a country where WCPayments is eligible.

**Event tracking:**
I think we already have all the tracking in place, but let me know if I'm missing something.

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #32130

This PR hides `Additional ways to get paid` component and adds `Other payment methods` under the main payment method table when the store is located in WC Payments eligible country.

### How to test the changes in this Pull Request:

1. Start with a fresh install and complete OBW. Choose `United States` as your store country. Make sure not to install any extension in the `Business Details` section.
2. Navigate to `WooCommerce -> Settings -> Payments`
3. Confirm the `Additional ways to get paid` component is not visible.
4. Confirm `Other payment methods` link is shown.
5. `Navigate to `WooCommerce -> Settings -> General` and change the store country to Mexico (WC Payments is not available in Mexico).
6. Navigate to `WooCommerce -> Settings -> Payments` again and confirm `Additional ways to get paid` component is visible.
7. Click `Other payment methods`
8. Wait for 5~10 mins. Go to live track and search for `wcadmin_settings_payments_recommendations_other_options`

### Screenshots
Screen Shot 2022-03-31 at 1 35 49 PM

### Changelog entry

> Adds Other payment methods link to the payment setting page when the store is located in WC Payments eligible country.

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by adrianduffell — DISMISSED** [review]
LGTM and tested perfectly, thank you @moon0326! 

It seems like it just needs some unit tests updated. Pre-approving 🚀

**Comment by ilyasfoo** [comment]
@moon0326 I noticed there's no track for clicking the new button but @pmcpinto mentioned tracking in the issue. I think we may need to add a new one if @pmcpinto can confirm it's needed

**Comment by pmcpinto** [comment]
> @moon0326 I noticed there's no track for clicking the new button but @pmcpinto mentioned tracking in the issue. I think we may need to add a new one if @pmcpinto can confirm it's needed

Thanks for the heads up. Yes, we should add a Tracks event to this call to action. Something like: wcadmin_settings_payments_recommendations_other_options

Is it possible to include a prop listing all the methods displayed in the table?

**Comment by moon0326** [comment]
> Is it possible to include a prop listing all the methods displayed in the table?

Yes, it's possible. I'll include them.

**Comment by moon0326** [comment]
> @moon0326 I noticed there's no track for clicking the new button but @pmcpinto mentioned tracking in the issue. I think we may need to add a new one if @pmcpinto can confirm it's needed

Thank you for the catch 👍

**Comment by moon0326** [comment]
@ilyasfoo @adrianduffell Added track in https://github.com/woocommerce/woocommerce/pull/32447/commits/4f633ddfdbeafdd3027cfc729786c41e5dfcad18 and https://github.com/woocommerce/woocommerce/pull/32447/commits/1c4234245352092d667c2c3c3268c466e2fd8968

**Comment by ilyasfoo** [comment]
Thanks, @moon0326! <del>I tested this but I wasn't able to get the tracks recorded. I found out in my browser's network tab that the `settings.js` loaded is from `http://localhost:8888/wp-content/plugins/woocommerce/assets/js/admin/settings.js?ver=6.4.0` instead of the legacy one. I'm not sure how to get the legacy one 🤔 </del>

Edit: My bad, I needed to build WC to get it.

I noticed some GitHub checks are failing too, maybe try to rebase with trunk to resolve that

File: plugins/woocommerce/legacy/js/admin/settings.js [hunk 1]
```diff
@@ -179,5 +179,20 @@
 			}
 		} );
 
+		$('#settings-other-payment-methods').on('click', function( e ) {
+			if ( typeof window.wcTracks.recordEvent === 'undefined' ) {
```
Comment (ilyasfoo) [inline plugins/woocommerce/legacy/js/admin/settings.js]: Nitpicky but we can actually use `window.wc.tracks.recordEvent` as well. Perhaps we can make it as first preference `const recordEvent = window.wc.tracks.recordEvent || window.wcTracks.recordEvent`. We can debug it better that way

**Review by ilyasfoo — DISMISSED** [review]
I've tested the new track and it works although it seems there's a slight delay between the UI finished loading and tracks setting up. It's possible for me to click on the button before tracks were set up and ended up not getting tracked.

However, I think there would be a very small number of users being able to read that fast before clicking the button, so I think it's not a deal breaker. Pre-approving!

File: plugins/woocommerce/legacy/js/admin/settings.js (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/legacy/js/admin/settings.js]: Thank you for the feedback! 👍 That's a good idea. I wasn't aware of the difference between wcTracks and wc.tracks. Updated in https://github.com/woocommerce/woocommerce/pull/32447/commits/bb3539b2e3f07909f27d95acd98473b7299c75cc

**Review by adrianduffell — APPROVED** [review]
The latest changes tested well for me. Thanks @moon0326 ! 🚀

**Pull request merged**